### PR TITLE
feat(transaction-summary): Endpoint for checking existence of web vit…

### PR DESCRIPTION
**Pull request by Zylphrex**
…als data

This adds a new endpoint to check for the existence of web vitals data. A transaction is said to have web vitals data if at least one event contains at least one of the web vitals. This endpoint only checks the last 7 days because

1. If a transaction starts receiving web vitals data, it's unlikely to stop receiving it, so checking the latest events is better.
2. To account for periods of low volume, and the fact that not all transactions will have web vitals, use a longer period to increase accuracy without scanning too much data.

Additionally, the results are cached for 5 minutes to avoid unnecessary queries. This isn't too long so that new events coming in will result in the web vitals tab showing not long after ingestion.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_has_measurements.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        "measurements.frames_frozen",
+        "measurements.stall_count",
+        "measurements.stall_stall_total_time",
+        "measurements.stall_stall_longest_time",
+    ],
+}
+
+
+class EventsHasMeasurementsQuerySerializer(serializers.Serializer):
+    transaction = serializers.CharField(max_length=200)
```
Comment (dashed) [inline src/sentry/api/endpoints/organization_events_has_measurements.py:37]: Would transaction and type be required?

File: src/sentry/api/endpoints/organization_events_has_measurements.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        # only allow one project at a time in order to cache the results
+        # for a unique transaction
+        project_ids = self.context.get("params", {}).get("project_id", [])
+        if len(project_ids) != 1:
+            raise serializers.ValidationError("Only 1 project allowed.")
+        return data
+
+
+class OrganizationEventsHasMeasurementsEndpoint(OrganizationEventsV2EndpointBase):
+    def get_snuba_params(self, request, organization, check_global_views=True):
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_has_measurements.py:48]: Does this endpoint need global-views? We could validate that they only pass a single project first then get snuba_params right

File: src/sentry/api/endpoints/organization_events_has_measurements.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        return params
+
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        with sentry_sdk.start_span(op="discover.endpoint", description="parse params"):
+            try:
+                params = self.get_snuba_params(request, organization)
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_has_measurements.py:71]: nit: couldn't we just do the date adjustments here instead?
```suggestion
                params = self.get_snuba_params(request, organization)
                now = timezone.now()
                params["start"] = now - timedelta(days=7)
                params["end"] = now
```

File: src/sentry/api/endpoints/organization_events_has_measurements.py (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_has_measurements.py:48]: Actually, good point, and I already validate they only passed a single project in the serializer above so the global views check can be safely removed.

File: src/sentry/api/endpoints/organization_events_has_measurements.py (hunk 1 above)
Comment (wmak) [reply src/sentry/api/endpoints/organization_events_has_measurements.py:35]: Actually should it be name + op?

Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_has_measurements.py:35]: Hm... why op?

Comment (wmak) [reply src/sentry/api/endpoints/organization_events_has_measurements.py:35]: for frontend measurements, they only exist on pageload ops, and by default we split the performance page between pageload&navigation

Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_has_measurements.py:35]: We ignore the op on the web vitals tab, so even if it has `transaction.op:navigation`, we overwrite it with `transaction.op:pageload` so I don't think this is a problem.

**Review by wmak — APPROVED** [review]

**Review by dashed — APPROVED** [review]

**Pull request merged**